hts221/hts221.go

```diff
@@ -0,0 +1,208 @@
+// Package hts221 implements a driver for HTS221,
+// a capacitive digital sensor for relative humidity and temperature.
+//
+// Datasheet: https://www.st.com/resource/en/datasheet/hts221.pdf
+//
+package hts221
+
+import (
+	"errors"
+	"machine"
+	"time"
+
+	"tinygo.org/x/drivers"
+)
+
+// Device wraps an I2C connection to a HTS221 device.
+type Device struct {
+	bus              drivers.I2C
+	Address          uint8
+	OnNano33BLE      bool
+	humiditySlope    float32
+	humidityZero     float32
+	temperatureSlope float32
+	temperatureZero  float32
+}
+
+// New creates a new HTS221 connection. The I2C bus must already be
+// configured.
+//
+// This function only creates the Device object, it does not touch the device.
+//
+// For Nano 33 BLE Sense, use machine.P0_15 (SCL1) and machine.P0_14 (SDA1),
+// and set onNano33BLE as hts221.ON_NANO_33_BLE.
+func New(bus drivers.I2C, deviceType uint8) Device {
+	return Device{bus: bus, Address: HTS221_ADDRESS, OnNano33BLE: deviceType == 1}
+}
+
+// Connected returns whether HTS221 has been found.
+// It does a "who am I" request and checks the response.
+func (d *Device) Connected() bool {
+
+	// if the HTS221 is on Nano 33 BLE Sense,
+	// turn on power pin (machine.P0_22) and I2C1 pullups power pin (machine.P1_00)
+	// and wait a moment.
+	if d.OnNano33BLE {
+		ENV := machine.Pin(22)
+		ENV.Configure(machine.PinConfig{Mode: machine.PinOutput})
+		ENV.High()
+		R := machine.Pin(32)
+		R.Configure(machine.PinConfig{Mode: machine.PinOutput})
+		R.High()
+		time.Sleep(time.Millisecond * 10)
+	}
```

Comment:
This should be done in the configuration step. A user is not required to call `.Connected()`.

Reply:
The trouble is that without pull high these pins first, the I2C bus simply hangs.

Would it be ok if I put it in ```New()``` (and I create two versions of them using build tags)?

Reply:
OK, just updated (including other drivers for Nano 33 BLE).